File: frontend_flutter/lib/modules/production/domain/entities/production_order.dart
/// Domain entity. Maps to backend ProductionOrderPayload.
class ProductionOrder {
  const ProductionOrder({
    required this.id,
    required this.orderNumber,
    required this.productId,
    required this.productCode,
    required this.productName,
    required this.productForm,
    required this.specification,
    required this.batchNumber,
    required this.productionDate,
    required this.expiryDate,
    required this.batchSizeLit,
    required this.quantitySpec1,
    required this.quantitySpec2,
    required this.status,
    required this.createdAt,
  });

  final String id;
  final String orderNumber;
  final String productId;
  final String productCode;
  final String productName;
  final String productForm;
  final String specification;
  final String batchNumber;
  final String productionDate;
  final String expiryDate;
  final double batchSizeLit;
  final int quantitySpec1;
  final int quantitySpec2;
  final String status;
  final String createdAt;
}

File: backend/internal/dto/production_order.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

// CreateProductionOrderRequest for POST production-orders.
// Business rule: 1 order = 1 product (ProductID is the only product).
type CreateProductionOrderRequest struct {
	ProductID       uuid.UUID `json:"productId"`
	BatchNumber     string    `json:"batchNumber"`
	ProductionDate  string    `json:"productionDate"`  // YYYY-MM-DD
	ExpiryDate      string    `json:"expiryDate"`      // YYYY-MM-DD
	BatchSizeLit    float64   `json:"batchSizeLit"`
	QuantitySpec1   int       `json:"quantitySpec1"`  // e.g. 100ml bottles
	QuantitySpec2   int       `json:"quantitySpec2"`  // e.g. 500ml bottles
	PlanItemID      *uuid.UUID `json:"planItemId,omitempty"` // optional link to production plan item
}

// UpdateProductionOrderRequest for PUT production-orders/:id.
type UpdateProductionOrderRequest struct {
	BatchNumber    string  `json:"batchNumber"`
	ProductionDate string  `json:"productionDate"`
	ExpiryDate     string  `json:"expiryDate"`
	BatchSizeLit   float64 `json:"batchSizeLit"`
	QuantitySpec1  int     `json:"quantitySpec1"`
	QuantitySpec2  int     `json:"quantitySpec2"`
}

// ProductionOrderPayload for list/get.
type ProductionOrderPayload struct {
	ID              uuid.UUID `json:"id"`
	OrderNumber     string    `json:"orderNumber"`
	ProductID       uuid.UUID `json:"productId"`
	ProductCode     string    `json:"productCode"`
	ProductName     string    `json:"productName"`
	ProductForm     string    `json:"productForm"`
	Specification   string    `json:"specification"`
	BatchNumber     string    `json:"batchNumber"`
	ProductionDate  time.Time `json:"productionDate"`
	ExpiryDate      time.Time `json:"expiryDate"`
	BatchSizeLit    float64   `json:"batchSizeLit"`
	QuantitySpec1   int       `json:"quantitySpec1"`
	QuantitySpec2   int       `json:"quantitySpec2"`
	Status          string    `json:"status"`
	CreatedAt       time.Time `json:"createdAt"`
	CreatedBy       uuid.UUID `json:"createdBy"`
}

// ProductionOrderListResponse for paginated list.
type ProductionOrderListResponse struct {
	Items  []ProductionOrderPayload `json:"items"`
	Total  int64                     `json:"total"`
	Limit  int                       `json:"limit"`
	Offset int                       `json:"offset"`
}
